cmd/spire-agent/cli/run/run.go

```diff
@@ -423,6 +426,24 @@ func NewAgentConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool)
 
 	ac.AllowUnauthenticatedVerifiers = c.Agent.AllowUnauthenticatedVerifiers
 
+	for _, authorizedDelegate := range c.Agent.AuthorizedDelegates {
+		_, err := common_cli.ParseTrustDomain(authorizedDelegate, logger)
+		if err != nil {
+			return nil, err
+		}
+
+		id, err := idutil.IDProtoFromString(authorizedDelegate)
+		if err != nil {
+			return nil, err
+		}
+
+		if id.TrustDomain != ac.TrustDomain.String() {
+			return nil, fmt.Errorf("authorized delegates should be in the same trust domain as the agent")
```

Comment:
I would say that they _must_ be in the same trust domain.
```suggestion
			return nil, fmt.Errorf("authorized delegates must be in the same trust domain as the agent")
```